### ISSUE TITLE
NumberFormatException in beagle

### ISSUE THREAD

**Issue by leiste375** [sbslee/pypgx#121]
Hi

The error previously described here (https://github.com/sbslee/pypgx/issues/113) seems to have resurfaced. Unfortunately I am unable to provide a VCF file as I'm not allowed to share the data in question. 

Is it possible that the function in utils.py, does not correctly process the error. I can see that it is caught as "IllegalErrorException"
            elif "IllegalArgumentException: 1.0" in message:
                warnings.warn("Beagle: Expectation-maximization skipped")
                vf3 = run_beagle(vf1, em='false')

However on my system (macOS, aarch64) it shows up as: "Exception in thread "main" java.lang.NumberFormatException".

I am happy to help out debugging in the following days.

**Comment by sbslee** [comment]
@leiste375,

Thanks for reporting this issue. Since I can't access your data, I will need some information from you to crack the problem. Could you please answer the following questions?

1. Which PyPGx version are you currently using?
2. Could you post the entire error message from Beagle?
3. I'm curious how you came to the conclusion that the error you encountered is the same one in #113. Have you already tried to turn off the expectation–maximization option (i.e., `em=false`) and the error disappeared? As far as I can tell, the error messages are different so it could be a totally different kind of problem.

**Comment by leiste375** [comment]
I apologize you are correct. I am currently using pypgx 0.23, and first saw the error appear after updating my environment. I saw that a similar issue was fixed in 0.22 and jumped to conclusions. However, setting em to false does not help here. 

After further testing I realize that it's related to a change in my dataset itself, and it appears in all versions I've tested (0.21-23).

It also does not appear consistently for all genes or even the same gene across different files.

Here is the full error message:
Exception in thread "main" java.lang.NumberFormatException
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:542)
	at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:567)
	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:670)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:927)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at vcf.VcfIt.fillEmissionBuffer(VcfIt.java:247)
	at vcf.VcfIt.<init>(VcfIt.java:226)
	at vcf.VcfIt.create(VcfIt.java:199)
	at vcf.VcfIt.create(VcfIt.java:174)
	at vcf.RefTargSlidingWindow.targIt(RefTargSlidingWindow.java:94)
	at vcf.RefTargSlidingWindow.<init>(RefTargSlidingWindow.java:80)
	at vcf.RefTargSlidingWindow.instance(RefTargSlidingWindow.java:70)
	at main.Main.slidingWindow(Main.java:129)
	at main.Main.main(Main.java:107)

Caused by: java.lang.NumberFormatException: For input string: "1/2"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
	at java.base/java.lang.Integer.parseInt(Integer.java:661)
	at java.base/java.lang.Integer.parseInt(Integer.java:777)
	at vcf.VcfRecGTParser.parseAllele(VcfRecGTParser.java:451)
	at vcf.VcfRecGTParser.hapListRep(VcfRecGTParser.java:312)
	at vcf.VcfIt.lambda$static$0(VcfIt.java:77)
	at vcf.VcfIt.lambda$new$3(VcfIt.java:220)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.IntPipeline$1$1.accept(IntPipeline.java:180)
	at java.base/java.util.stream.Streams$RangeIntSpliterator.forEachRemaining(Streams.java:104)
	at java.base/java.util.Spliterator$OfInt.forEachRemaining(Spliterator.java:712)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:960)
	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:934)
	at java.base/java.util.stream.AbstractTask.compute(AbstractTask.java:327)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)

**Comment by sbslee** [comment]
@leiste375,

The offending string is `"1/2"` in `Caused by: java.lang.NumberFormatException: For input string: "1/2"`, which represents a genotype call heterozygous for two different alternative alleles. Beagle was expecting to receive a number but it was given this string. My gut feeling tells me that this is likely the result of an incorrectly formatted VCF file. But since I can't see your data, it's difficult to be certain.

Could you check to see if you are able to follow the [GeT-RM WGS tutorial](https://pypgx.readthedocs.io/en/latest/tutorials.html#get-rm-wgs-tutorial)? If so, can you try to compare your input VCF file to the input files used in the tutorial? This may give you a clue as to why Beagle is having an issue processing your VCF.

**Comment by leiste375** [comment]
Yes, I agree. I lack information regarding the pipeline itself, and can only infer some info from the header. I only have access to the VCFs, and none of the prior data.

I will close the issue. I opened it, thinking I had spotted a quick and easy fix. 
If I can get my hands on more data and information, I might re-open it, or open a new issue but as it stands it's probably not worth the time fully debugging it. 

In any case, thank you for the taking the time and your quick response!
